src/libs/actions/Report.js

```diff
@@ -458,9 +468,9 @@ function getReportChannelName(reportID) {
 }
 
 /**
- * Initialize our pusher subscriptions to listen for new report comments
+ * Initialize our pusher subscriptions to listen for new report comments and pin toggles
  */
-function subscribeToReportCommentEvents() {
+function subscribeToReportCommentAndTogglePinnedEvents() {
```

Comment:
Originally, I tried implementing a new function, `subscribeToReportTogglePinnedEvents` instead of subscribing to `Pusher.TYPE.REPORT_TOGGLE_PINNED` inside this function, but this would mean line 69 in `src/libs/Navigation/AppNavigator/AuthScreens.js` would have to be something like:

```js
        }).then(() => {
            subscribeToReportCommentEvents();
            subscribeToReportTogglePinnedEvents();
        });
```

And if we do this, in `subscribeToReportTogglePinnedEvents` we would probably want to check if the `REPORT_TOGGLE_PINNED` event was already subscribed to in that account channel (`private-user-accountID-${currentUserAccountID}/`;) - otherwise, don't subscribe again
- Similar to how we currently do `if (Pusher.isSubscribed(pusherChannelName) || Pusher.isAlreadySubscribing(pusherChannelName)) {` and return early if we're already subscribed to that account channel.

Pusher has an easy way to check if you're subscribed to a channel (see `.isSubscribed` in `src/libs/Pusher/pusher.js`), but I couldn't find a way to check if that channel was already bound to a specific event or not.

We could keep track of that in Onyx or potentially in a local variable in `pusher.js` (like how we keep track of `socket` there... But I thought that would be a bit ugly. It was easier to just combine the two subscribe functions.

Hope this makes sense 😅

Reply:
So is the idea that this would be a general subscribing method for the `private-user-accountID-*` channel? If so, should we call this `subscribeToUserEvents`?

Can you explain how you envision it would work if we kept track of it in Onyx? I'm not sure I follow that part.

Reply:
1. Love the idea of renaming the function `subscribeToUserEvents` if we go with the solution of having one function handling all user-account channels 👍 

2. Yeah for the possible Onyx integration I was thinking you could subscribe to some new key like `PUSHER_SUBSCRIBED_EVENTS`:

```js
let pusherSubscribedEvents = [];
Onyx.connect({
    key: ONYXKEYS.PUSHER_SUBSCRIBED_EVENTS,
    callback: val => pusherSubscribedEvents = val;
});
```

And any time you subscribe to a channel, you'd add an Onyx.merge call like this:

```js
function subscribeToReportCommentEvents() {
    // NEW: instead of checking if channel has been subscribed to, check if event has been subscribed to
    if (pusherSubscribedEvents.includes(Pusher.TYPE.REPORT_COMMENT)) {
        return;
    }

    // We're "about to subscribe" to this event, so add to onyx so we won't subscribe to this event in the future
    Onyx.merge(ONYXKEYS.PUSHER_SUBSCRIBED_EVENTS, [Pusher.TYPE.REPORT_COMMENT]);

    Pusher.subscribe(pusherChannelName, Pusher.TYPE.REPORT_COMMENT, (pushJSON) => {
        // ...
    }).catch(() => {
        // Here, we'd remove Pusher.TYPE.REPORT_COMMENT from Onyx's subscribed events
    })
}

function subscribeToReportTogglePinnedEvents() {
    // ... similar to code above
}
```

Reply:
Updated function name to `subscribeToUserEvents` 👍

Reply:
Ah, seems like that might be over-engineering a bit - fine with it the way it is.

Reply:
Yeah I agree, but thanks for thinking about the idea with me 👍